### PR TITLE
Management of in-mesh certificates

### PR DESCRIPTION
**Pull request by jacob-delgado**
Help users manage in-mesh certificates. It is difficult for users to understand how to use and change these settings.

While it is environment variables

Related to https://github.com/istio/istio/pull/45382 and https://github.com/istio/istio/pull/44459 (and various other PRs)

This was a request from @costinm. See https://github.com/istio/istio/pull/44459#pullrequestreview-1406168391

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

- [ ] Ambient
- [ ] Docs
- [X] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [X] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: content/en/docs/ops/configuration/traffic-management/manage-mesh-certificates/index.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+Changing this will require a istiod modifying either the IOP file you give to
+istioctl or the values file you pass into your Helm install for the istio-discovery chart.
+
+{{< tip >}}
+While there are many environment variables that can be changed for
+[pilot-discovery](/docs/reference/commands/pilot-discovery/) this document will only
+outline some of them.
+{{< /tip >}}
+
+{{< tabset category-name="gateway-install-type" >}}
```
Comment (jacob-delgado) [inline content/en/docs/ops/configuration/traffic-management/manage-mesh-certificates/index.md:43]:
```suggestion
{{< tabset category-name="certificates" >}}
```

File: content/en/docs/ops/configuration/traffic-management/manage-mesh-certificates/index.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+## Sidecars
+
+Since sidecars manage their own certificates for in-mesh communication, the sidecars
+are responsible for managing their private keys and generated CSRs. The sidecar
+injector needs to be modified to inject the environment variables to be used for
+this purpose.
+
+{{< tip >}}
+While there are many environment variables that can be changed for
+[pilot-agent](/docs/reference/commands/pilot-agent/) this document will only
```
Comment (windsonsea) [inline content/en/docs/ops/configuration/traffic-management/manage-mesh-certificates/index.md:98]:
```suggestion
[pilot-agent](/docs/reference/commands/pilot-agent/), this document will only
```

File: content/en/docs/ops/configuration/traffic-management/manage-mesh-certificates/index.md [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+      ...
+{{< /text >}}
+
+{{< /tab >}}
+
+{{< /tabset >}}
+
+### Signature Algorithm
+
+By default, the sidecars will create RSA based certificates. If the user wishes to change it to
```
Comment (windsonsea) [inline content/en/docs/ops/configuration/traffic-management/manage-mesh-certificates/index.md:155]:
```suggestion
By default, the sidecars will create RSA-based certificates. If you want to change it to
```

File: content/en/docs/ops/configuration/traffic-management/manage-mesh-certificates/index.md [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+ECC then `ECC_SIGNATURE_ALGORITHM` must be set to `ECDSA`
```
Comment (windsonsea) [inline content/en/docs/ops/configuration/traffic-management/manage-mesh-certificates/index.md:156]:
```suggestion
ECC, you need to set `ECC_SIGNATURE_ALGORITHM` to `ECDSA`.
```

File: content/en/docs/ops/configuration/traffic-management/manage-mesh-certificates/index.md [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+      ECC_SIGNATURE_ALGORITHM: "ECDSA"
+{{< /text >}}
+
+{{< /tab >}}
+
+{{< /tabset >}}
+
+Only P256 and P384 are supported via `ECC_CURVE`.
+
+To modify the RSA key size if you wish to continue using RSA signature algorithms
```
Comment (windsonsea) [inline content/en/docs/ops/configuration/traffic-management/manage-mesh-certificates/index.md:189]:
```suggestion
If you prefer to retain RSA signature algorithms and want to modify the RSA key size,
```

File: content/en/docs/ops/configuration/traffic-management/manage-mesh-certificates/index.md [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+you can change `WORKLOAD_RSA_KEY_SIZE`.
```
Comment (windsonsea) [inline content/en/docs/ops/configuration/traffic-management/manage-mesh-certificates/index.md:190]:
```suggestion
you can change the value of `WORKLOAD_RSA_KEY_SIZE`.
```

**Review by windsonsea — COMMENTED** [review]
Polish the text for easy understanding.

**Comment by windsonsea** [comment]
Kindly reminder: need to resolve the lint error.

File: content/en/docs/ops/configuration/traffic-management/manage-mesh-certificates/index.md [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+      ...
+{{< /text >}}
+
+{{< /tab >}}
+
+{{< /tabset >}}
+
+### Signature Algorithm
+
+By default, the sidecars will create RSA-based certificates. If you want to change it to
```
Comment (craigbox) [inline content/en/docs/ops/configuration/traffic-management/manage-mesh-certificates/index.md:160]: I think this line needs `RSA-based` added to `.spelling`, and that just makes me angry.

**Review by craigbox — CHANGES_REQUESTED** [review]

**Pull request closed without merging**